src/graph/expression_operators.cpp

```diff
@@ -546,13 +548,16 @@ Expr affine(Expr a, Expr b, Expr bias, bool transA, bool transB, float scale) {
   Type bElementType = b->value_type();
 
   if(device == DeviceType::cpu) {
-    if(isFloat(aElementType) && (isFloat(bElementType) || isIntgemm(bElementType))) {
-      if(a->graph()->getBackend()->isInt8()  || matchType<intgemm8>(bElementType) ) {
+    if(isFloat(aElementType) && isFloat(bElementType)) {
+      return affineDefault(a, b, bias, transA, transB, scale);
+    } else if(isFloat(aElementType) && isIntgemm(bElementType)) {
+      // @TODO: this branch should move into cpu::integer::*
+      if(sizeOf(bElementType) == 1) {
         return cpu::integer::affine<Type::int8>(a, b, bias, transA, transB, scale, clipValue);
-      } else if(a->graph()->getBackend()->isInt16()  || matchType<intgemm16>(bElementType) ) {
+      } else if(sizeOf(bElementType) == 2) {
```

Comment:
I don't understand this, why not use matchtype? It's not obvious what `sizeOf(bElementType) == 2` means

Reply:
You would need to list all 8 types here. See explanation below.